### ISSUE TITLE
[BB pr#229] style(global): fix eslint

### ISSUE THREAD

**Issue by coveobot** [coveo/ui-kit#229]
>  **Pull request** :twisted_rightwards_arrows: created by **@louis-bompart** on 2020-11-02 23:44
> Last updated on 2020-11-03 20:53
> Original Bitbucket pull request id: 229
>
> Participants:
>
> * **@acote-coveo** (reviewer)
> * **@ThibodeauJF** (reviewer) :heavy_check_mark:
> * **@btaillon** (reviewer)
> * bitbucket user **coveo-anti-throttling_02**
> * **@jlblais** (reviewer)
> * **@olamothe** (reviewer)
> * **@samisayegh** (reviewer) :heavy_check_mark:
> * **@nrawji** (reviewer)
>
> Source: Commit `5a1599c49947` on branch `KIT-222`> Destination: https://bitbucket.org/coveord/ui-kit/commits/d565c44250b7 on branch master
> Merge commit: https://github.com/None/commit/1f69ac1cbe91
>
> State: **`MERGED`**

# Minimal Changes:

Rule of thumb followed what was present. If a rule was tripping a thousand errors and was not a ‘basic’ one \(like unused-var\), I ignored it \(you can see there’s very few of them\).

* Use ESLint as a lint-runner instead of GTS.
* Use GTS ruleset and apply it only on TypeScript
* Use `eslint:recommended` for all files.
* Bump-up ecmaVersion and make all files ‘module’ \(allows import\)
* `.prettierrc.json` → `.prettierrc.js` to allow extendability \(used for LWC template, see `packages/quantic/.prettierrc.js`\)

# ‘Ease-of-use’ Changes:

Those changes are more to ease the linter pain on our day-to-day:

* `@typescript-eslint/no-unused-vars`, ‘ignoreRestSibling' set to true to allow object destructuring in TypeScript. Was not used in JavaScript so did not enable it \(see ‘Next improvements’\).
* `no-unused-vars`, `argsIgnorePattern` set to `^_`: allows an unused variable if prefixed by an underscore. This can make sense in functions where the first argument is not used \(e.g. cypress hooks\).

# Next improvements:

* By updating `@typescript-eslint` shenanigans, `@typescript-eslint/explicit-module-boundary-types` and `@typescript-eslint/ban-types` tripped a lot, which is normal because they were not there when the code was written.  
  Because the code does not follow those rules right now, I disabled them for now.  
  I personally think that those rules make some kind of sense and there’d be a good addition.  
  However, to keep this PR tidy, I would consider enabling those rules in another PR.
* Indeed, I’d say that a linter ruleset should be evolutive. So if when you type code you’re blocked by the linter for what you consider “stupid”, it might be the moment to take a step back and consider why it’s doing it and either tweak the ruleset, or your code. On the other side, when you review code, being it yours or someone else's if you find something that you think the linter should have caught, then check if a rule exists for it and add it to the ruleset.

[https://coveord.atlassian.net/browse/KIT-222](https://coveord.atlassian.net/browse/KIT-222)

‌
